atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbErrorDecoder.java

```diff
@@ -27,6 +27,8 @@
 import feign.codec.ErrorDecoder;
 
 public class AtlasDbErrorDecoder implements ErrorDecoder {
+    private static final String LOCK_METHOD_KEY = "RemoteLockService#lock(String,LockRequest)";
```

Comment:
Hmm, this is a little fragile. Also not sure that we want to retry on all 500 errors (at least in the scope of this PR).

It may be better to catch `InterruptedException` server-side and map it to a 503. I would not be opposed to having a check in `AwaitingLeadershipProxy` that checks if the delegate throws interrupted exception, and if we are no longer the leader propagates it as `NotCurrentLeaderException`